EntityModel/ContractResolver.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Newtonsoft.Json.Serialization;
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+
+namespace EntityModel
+{
+    public class ContractResolver<T> : DefaultContractResolver
```

Comment:
This allows shared fields like ID to be in a base class while still having a different JSON name in each derived class